src/internal/item-container/index.tsx

```diff
@@ -296,7 +298,7 @@ function ItemContainerComponent(
     // 1. If the last interaction is not "keyboard" (the user clicked on another handle issuing a new transition);
     // 2. If the item is borrowed (in that case the focus moves to the acquired item which is expected).
     if (transition && transition.interactionType === "keyboard" && !transition.isBorrowed) {
```

Comment:
I think the transition.isBorrowed check can be removed now. The submitTransition should be ignored by the controller once the sourceId is replaced.